sql: TestStatementCancelRollback failed

sql.TestStatementCancelRollback [failed](https://mesolite.cluster.engflow.com/invocations/default/ec1e8d54-841c-411b-9d0d-39cbac78bf6f?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [c7e7b1997b7d39342bc22fc7f2dfd3df301c7d4f](https://github.com/cockroachdb/cockroach/commits/c7e7b1997b7d39342bc22fc7f2dfd3df301c7d4f):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m54s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=false (14m53s)

```
Stack: 

```
goroutine 55718 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStatementCancelRollback4041603945
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=false
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40247

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16036882?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16036882?buildTab=artifacts#/) on master @ [c7e7b1997b7d39342bc22fc7f2dfd3df301c7d4f](https://github.com/cockroachdb/cockroach/commits/c7e7b1997b7d39342bc22fc7f2dfd3df301c7d4f):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m51s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=true (14m51s)

```
Stack: 

```
goroutine 55453 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback1665463314
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=true
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16038579?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16038579?buildTab=artifacts#/) on master @ [5070f16a5ca870ad4b9459a14b148ddb4a6b6fa0](https://github.com/cockroachdb/cockroach/commits/5070f16a5ca870ad4b9459a14b148ddb4a6b6fa0):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m52s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=true (14m52s)

```
Stack: 

```
goroutine 55186 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback683401664
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=true
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16039546?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16039546?buildTab=artifacts#/) on master @ [5070f16a5ca870ad4b9459a14b148ddb4a6b6fa0](https://github.com/cockroachdb/cockroach/commits/5070f16a5ca870ad4b9459a14b148ddb4a6b6fa0):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m54s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=true (14m54s)

```
Stack: 

```
goroutine 55276 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback1798106609
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=true
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16039768?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16039768?buildTab=artifacts#/) on master @ [1bf1d23fe7af1001d2bfffe20def2fca3412c750](https://github.com/cockroachdb/cockroach/commits/1bf1d23fe7af1001d2bfffe20def2fca3412c750):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m53s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=false (14m51s)

```
Stack: 

```
goroutine 57049 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback821018415
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=false
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16041955?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16041955?buildTab=artifacts#/) on master @ [272308f2498890129ab370f8a67d57ac26d9d4f3](https://github.com/cockroachdb/cockroach/commits/272308f2498890129ab370f8a67d57ac26d9d4f3):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m53s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=true (14m53s)

```
Stack: 

```
goroutine 55213 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback2008117952
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=true
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16042652?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16042652?buildTab=artifacts#/) on master @ [272308f2498890129ab370f8a67d57ac26d9d4f3](https://github.com/cockroachdb/cockroach/commits/272308f2498890129ab370f8a67d57ac26d9d4f3):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m54s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=true (14m54s)

```
Stack: 

```
goroutine 55277 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback3901216413
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=true
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://tanzanite.cluster.engflow.com/invocations/default/adb84329-fd75-46ae-9a59-ee208d9f9db6?testReportRun=8&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [3afb6935d0ef3de7c0d44cfb3cd54f312752c186](https://github.com/cockroachdb/cockroach/commits/3afb6935d0ef3de7c0d44cfb3cd54f312752c186):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m54s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=false (14m53s)

```
Stack: 

```
goroutine 55600 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStatementCancelRollback3912552370
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=false
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16047009?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16047009?buildTab=artifacts#/) on master @ [903fd7169cec06261630b1cb000355618766c103](https://github.com/cockroachdb/cockroach/commits/903fd7169cec06261630b1cb000355618766c103):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m50s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=false (14m48s)

```
Stack: 

```
goroutine 57303 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback3028701194
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=false
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16049637?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16049637?buildTab=artifacts#/) on master @ [23ab2790ad7978b4dfcb117a542ee461eb127a19](https://github.com/cockroachdb/cockroach/commits/23ab2790ad7978b4dfcb117a542ee461eb127a19):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m52s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=true (14m52s)

```
Stack: 

```
goroutine 32747 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback2453387605
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=true
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16050801?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16050801?buildTab=artifacts#/) on master @ [23ab2790ad7978b4dfcb117a542ee461eb127a19](https://github.com/cockroachdb/cockroach/commits/23ab2790ad7978b4dfcb117a542ee461eb127a19):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m55s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=false (14m53s)

```
Stack: 

```
goroutine 57541 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback1153398505
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=false
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16051478?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16051478?buildTab=artifacts#/) on master @ [23ab2790ad7978b4dfcb117a542ee461eb127a19](https://github.com/cockroachdb/cockroach/commits/23ab2790ad7978b4dfcb117a542ee461eb127a19):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m54s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=true (14m54s)

```
Stack: 

```
goroutine 32537 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback2880918179
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=true
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestStatementCancelRollback [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16053992?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16053992?buildTab=artifacts#/) on master @ [312353a82d90672d383318eeaaa5d4fef20cc62e](https://github.com/cockroachdb/cockroach/commits/312353a82d90672d383318eeaaa5d4fef20cc62e):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestStatementCancelRollback (14m50s)
	TestStatementCancelRollback/Cancel_with_statement_timeout=false (14m48s)

```
Stack: 

```
goroutine 35067 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStatementCancelRollback
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestStatementCancelRollback1078145853
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStatementCancelRollback/Cancel_with_statement_timeout=false
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatementCancelRollback.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/conn_executor_test.go
+++ b/pkg/sql/conn_executor_test.go
@@ -790,9 +790,8 @@ func TestRetriableErrorDuringPrepare(t *testing.T) {
 func TestStatementCancelRollback(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	// This test is on the slower side, so skip under duress.
 	skip.UnderDuress(t)
-	// Flaky test.
-	skip.WithIssue(t, 127032)
 
 	for _, useStatementTimeout := range []bool{true, false} {
 		t.Run(fmt.Sprintf("Cancel with statement timeout=%t", useStatementTimeout),
@@ -806,7 +805,7 @@ func TestStatementCancelRollback(t *testing.T) {
 				if useStatementTimeout {
 					queryCtx, cancelFn = context.WithCancel(ctx)
 				} else {
-					queryCtx, cancelFn = context.WithTimeout(ctx, 1*time.Second)
+					queryCtx, cancelFn = context.WithTimeout(ctx, 10*time.Second)
 				}
 				defer cancelFn()
 				s, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{
@@ -814,8 +813,8 @@ func TestStatementCancelRollback(t *testing.T) {
 						Store: &kvserver.StoreTestingKnobs{
 							TestingRequestFilter: func(ctx context.Context, request *kvpb.BatchRequest) *kvpb.Error {
 								// Once the hook is enabled we are expecting a txn rollback involving
-								// the system.descriptor key, because the first tihng accessed by the
-								// txn below is the descriptor table.
+								// the system.descriptor / system.namespace key, because the first thing
+								// accessed by the txn one of those tables (depending on if retries occur).
 								if hookEnabled.Load() {
 									if request.IsSingleEndTxnRequest() {
 										if !request.Requests[0].GetEndTxn().Commit {
@@ -823,7 +822,7 @@ func TestStatementCancelRollback(t *testing.T) {
 											if err != nil {
 												return nil
 											}
-											if tblID == keys.DescriptorTableID {
+											if tblID == keys.DescriptorTableID || tblID == keys.NamespaceTableID {
 												// This channel will only be closed once the "synchronous" rollback returns.
 												<-rollbackExpected
 												close(rollbackCompleted)
@@ -847,7 +846,7 @@ func TestStatementCancelRollback(t *testing.T) {
 					_, err = conn.ExecContext(queryCtx, "SET statement_timeout='1s'")
 					require.NoError(t, err)
 				}
-				_, err = conn.ExecContext(queryCtx, "CREATE TABLE t1(n int);SELECT * FROM pg_sleep(5)")
+				_, err = conn.ExecContext(queryCtx, "CREATE TABLE t1(n int);SELECT * FROM pg_sleep(20)")
 				expectedError := "query execution canceled due to statement timeout"
 				if !useStatementTimeout {
 					expectedError = "pq: query execution canceled"
@@ -857,11 +856,15 @@ func TestStatementCancelRollback(t *testing.T) {
 					expectedError,
 					"expected timeout error")
 				// Because the rollback is asynchronous due to the timeout, we expected
-				// to just return here. Any rollbacks involving the descriptor key above are
-				// *blocked*.
+				// to just return here. Any rollbacks involving the descriptor/namespace key a
+				// above are  *blocked*.
 				close(rollbackExpected)
 				// Confirm the async rollback happened.
-				<-rollbackCompleted
+				select {
+				case <-rollbackCompleted:
+				case <-time.After(time.Minute):
+					t.Fatal("dead")
+				}
 			})
 	}
 }

